tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php

```diff
@@ -11,42 +11,14 @@
 
 class GetsWalletCryptocurrenciesControllerTest extends TestCase
 {
-    private WalletDataSource $walletDataSource;
-
-    protected function setUp(): void
-    {
-        parent::setUp();
-        $this->walletDataSource = Mockery::mock(WalletDataSource::class);
-        $this->app->bind(WalletDataSource::class, function () {
-            return $this->walletDataSource;
-        });
-    }
-
     /**
      * @test
      */
-    public function ifBadWalletIdThrowsBadRequest()
-    {
-        $this->walletDataSource
-            ->expects("findById")
-            ->with(null)
-            ->times(0)
-            ->andReturn(null);
-
-        $response = $this->get('api/wallet/-5');
-        $response->assertBadRequest();
-    }
-
-    /**
-     * @test
-     */
-    public function ifWalletIdNotFoundThrowsError()
+    public function returnsNotFoundMessageWhenWalletDoesNotExist()
```

Comment:
You don´t need to start all the tests with returns. Think about what are you going to test.
If you´re going to test that a wallet was not found if wallet does not exists.
```suggestion
    public function walletIdWasNotFoundIfWalletDoesNotExists()
```